app/assets/javascripts/flood_risk_engine/application.js

```diff
@@ -12,3 +12,11 @@
 //
 //= details.polyfill
 //= require_tree .
+
+
+$(function() {
```

Comment:
this is solvable with JS, not sure how this translates to your state machine but we were able to ID from the params when Back had been clicked and then work out what the available  :back transition was for the enrollment and use that transition to step back to the right state .. code for the main routine

```
    def validate_state_pre_edit
      # we bust the browser cache to ensure forms are reloaded but this causes Browser back button to behave
      # oddly or not at all
      # So we need to identify when user has gone back one state, without rendering the form
      # while trying to keep checks/code called down to a minimum

      if(params[:rendered_state].nil? && params[:state] && back_button_param_list.all? {|p| params.key?(p) })

        transitions = @enrollment.transitions_for

        back_event = transitions.find { |t| t.event == :back }

        if(back_event && back_event.from == @enrollment.state && back_event.to == params[:state])
          logger.debug("User CLICKED back Button - resetting state to [#{params[:state]}] from #{@enrollment.state}")
          @enrollment.update!(under_review: false) if(@enrollment.under_review?)
          @enrollment.update!(state: params[:state])
        end

      end
    end
```